pkg/clusteragent/admission/controllers/webhook/controller_base.go

```diff
@@ -189,7 +189,7 @@ func generateAutoInstrumentationWebhook(wmeta workloadmeta.Component, datadogCon
 		return nil, fmt.Errorf("failed to create auto instrumentation config: %v", err)
 	}
 
-	apm, err := autoinstrumentation.NewMutatorWithFilter(config, wmeta, imageResolver)
+	apm, err := autoinstrumentation.NewTargetMutator(config, wmeta, imageResolver)
```

Comment:
One thing to consider is that we can have the NewMutatorWithFilter do everything that `NewTargetMutator` was doing (renaming that function) to keep the scope of this change to be _internal_ to the autoinstrumentation package and make it so that the webhook registration _outside_ of that stays exactly the same.